suite-common/message-system/src/messageSystemUtils.ts

```diff
@@ -234,76 +232,94 @@ export const getValidMessages = (config: MessageSystem | null, options: Options)
-                        settings: settingsCondition,
-                        devices: deviceCondition,
-                    } = condition;
-
-                    if (durationCondition && !validateDurationCompatibility(durationCondition)) {
-                        return false;
-                    }
-
-                    if (
-                        environmentCondition &&
-                        !validateEnvironmentCompatibility(
-                            environmentCondition,
-                            environment,
-                            suiteVersion,
-                            commitHash,
-                        )
-                    ) {
-                        return false;
-                    }
-
-                    if (
-                        osCondition &&
-                        !validateVersionCompatibility(osCondition, currentOsName, currentOsVersion)
-                    ) {
-                        return false;
-                    }
-
-                    if (
-                        environment === 'web' &&
-                        browserCondition &&
-                        !validateVersionCompatibility(
-                            browserCondition,
-                            currentBrowserName,
-                            currentBrowserVersion,
-                        )
-                    ) {
-                        return false;
-                    }
-
-                    if (
-                        settingsCondition &&
-                        !validateSettingsCompatibility(settingsCondition, settings)
-                    ) {
-                        return false;
-                    }
-
-                    if (
-                        transportCondition &&
-                        !validateTransportCompatibility(transportCondition, transport)
-                    ) {
-                        return false;
-                    }
-
-                    if (deviceCondition && !validateDeviceCompatibility(deviceCondition, device)) {
-                        return false;
-                    }
-
-                    return true;
-                }),
+                action.conditions.some(condition => validateConditions(condition, options)),
         )
         .map(action => action.message);
 };
+
+export const validateExperiment = (experiment: ExperimentsItem) =>
+    experiment.groups.length >= 2 &&
+    experiment.groups.reduce((acc, item) => acc + item.percentage, 0) === 100;
+
+export const getValidExperiments = (
+    config: MessageSystem | null,
+    options: Options,
+): ExperimentsItem[] => {
+    if (!config?.experiments) {
+        return [];
+    }
+
+    return config.experiments
+        .filter(
+            experiment =>
+                !experiment.conditions.length ||
+                experiment.conditions.some(condition => validateConditions(condition, options)),
+        )
+        .filter(experiment => validateExperiment(experiment.experiment))
```

Comment:
Is this filter needed here? This should be ensured by the scheme itself. Here I think we want to be as benevolent as possible.

Reply:
Of course `experiment.groups.length >= 2 &&` is ambiguous, but the schema does not handle control of the sum. Or is there a way how to do it in the schema?